Error while compiling bash-interactive static for bootstrap tools with the first configure flag removed:

``
error: builder for '/nix/store/lq6m48dxw21l9vamc3dbhr4k2ffxs68f-bash-x86_64-unknown-freebsd-5.2p26.drv' failed with exit code 2;
       last 25 log lines:
       >       | ^
       > 19 warnings generated.
       > rm -f bash
       > x86_64-unknown-freebsd-clang -L./builtins -L./lib/readline -L./lib/readline -L./lib/glob -L./lib/tilde  -L./lib/sh   -g -O2   -o bash shell.o eval.o y.tab.o general.o make_cmd.o print_cmd.o  dispose_cmd.o execute_cmd.o variables.o copy_cmd.o error.o expr.o flags.o jobs.o subst.o hashcmd.o hashlib.o mailcheck.o trap.o input.o unwind_prot.o pathexp.o sig.o test.o version.o alias.o array.o arrayfunc.o assoc.o braces.o bracecomp.o bashhist.o bashline.o  list.o stringlib.o locale.o findcmd.o redir.o pcomplete.o pcomplib.o syntax.o xmalloc.o signames.o -lbuiltins -lglob -lsh  -lhistory  -ltilde  lib/intl/libintl.a     -ldl
       > x86_64-unknown-freebsd-ld: error: duplicate symbol: getenv
       > >>> defined at getenv.c:52
       > >>>            getenv.o:(getenv) in archive ./lib/sh/libsh.a
       > >>> defined at getenv.o:(.text+0xC0) in archive /nix/store/aalnvnpkv3h9a4zslbh39h0jc5fs8jf2-libc-x86_64-unknown-freebsd-14.0.0/lib/libc.a
       >
       > x86_64-unknown-freebsd-ld: error: duplicate symbol: setenv
       > >>> defined at getenv.c:161
       > >>>            getenv.o:(setenv) in archive ./lib/sh/libsh.a
       > >>> defined at getenv.o:(.text+0x280) in archive /nix/store/aalnvnpkv3h9a4zslbh39h0jc5fs8jf2-libc-x86_64-unknown-freebsd-14.0.0/lib/libc.a
       >
       > x86_64-unknown-freebsd-ld: error: duplicate symbol: putenv
       > >>> defined at getenv.c:107
       > >>>            getenv.o:(putenv) in archive ./lib/sh/libsh.a
       > >>> defined at getenv.o:(.text+0xE90) in archive /nix/store/aalnvnpkv3h9a4zslbh39h0jc5fs8jf2-libc-x86_64-unknown-freebsd-14.0.0/lib/libc.a
       >
       > x86_64-unknown-freebsd-ld: error: duplicate symbol: unsetenv
       > >>> defined at getenv.c:213
       > >>>            getenv.o:(unsetenv) in archive ./lib/sh/libsh.a
       > >>> defined at getenv.o:(.text+0x12C0) in archive /nix/store/aalnvnpkv3h9a4zslbh39h0jc5fs8jf2-libc-x86_64-unknown-freebsd-14.0.0/lib/libc.a
       > clang: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [Makefile:595: bash] Error 1
       For full logs, run 'nix-store -l /nix/store/lq6m48dxw21l9vamc3dbhr4k2ffxs68f-bash-x86_64-unknown-freebsd-5.2p26.drv'.
error: 1 dependencies of derivation '/nix/store/1fpaqmif194ff1g1ql4iddf6cycx178v-x86_64-freebsd-bash.drv' failed to build
```

--- a/pkgs/shells/bash/5.nix
+++ b/pkgs/shells/bash/5.nix
@@ -88,6 +88,10 @@ stdenv.mkDerivation rec {
     "bash_cv_termcap_lib=libncurses"
   ] ++ lib.optionals (stdenv.hostPlatform.libc == "musl") [
     "--disable-nls"
+  ] ++ lib.optionals stdenv.hostPlatform.isFreeBSD [
+    # /dev/fd is optional on FreeBSD. we need it to work when built on a system
+    # with it and transferred to a system without it! This includes linux cross.
+    "bash_cv_dev_fd=absent"
   ];
 
   strictDeps = true;

